File: app/src/androidTest/java/com/imdglobalservices/psi/PSISampleDataTest.java
package com.imdglobalservices.psi;

import com.imdglobalservices.psi.models.COEightHourMax;
import com.imdglobalservices.psi.models.COSubIndex;
import com.imdglobalservices.psi.models.Location;
import com.imdglobalservices.psi.models.NO2OneHourMax;
import com.imdglobalservices.psi.models.O3EightHourMax;
import com.imdglobalservices.psi.models.O3SubIndex;
import com.imdglobalservices.psi.models.PM10SubIndex;
import com.imdglobalservices.psi.models.PM10TwentyFourHourly;
import com.imdglobalservices.psi.models.PM25SubIndex;
import com.imdglobalservices.psi.models.PM25TwentyFourHourly;
import com.imdglobalservices.psi.models.PSITwentyFourHourly;
import com.imdglobalservices.psi.models.Region;
import com.imdglobalservices.psi.models.SO2SubIndex;
import com.imdglobalservices.psi.models.SO2TwentyFourHourly;

import java.util.ArrayList;

/**
 * @author Dwi Setiyono <dwi.setiyono@ovo.id>
 * @since 2017.15.11
 */
public class PSISampleDataTest {

    private ArrayList<Region> regionList = new ArrayList<>();

    protected void setTestData(double lat, double lng, String name) {
        Region region = new Region();
        Location location = new Location();
        location.setLatitude(lat);
        location.setLongitude(lng);
        region.setName(name);
        region.setLabelLocation(location);
        regionList.add(region);
    }

    protected O3SubIndex getO3SubIndexTest() {
        O3SubIndex o3SubIndex = new O3SubIndex();
        o3SubIndex.setEast(3);
        o3SubIndex.setCentral(4);
        o3SubIndex.setSouth(3);
        o3SubIndex.setNorth(5);
        o3SubIndex.setWest(4);
        o3SubIndex.setNational(5);
        return o3SubIndex;
    }

    protected PM10TwentyFourHourly getPM10TwentyFourHourly() {
        PM10TwentyFourHourly pm10TwentyFourHourly = new PM10TwentyFourHourly();
        pm10TwentyFourHourly.setEast(14);
        pm10TwentyFourHourly.setCentral(12);
        pm10TwentyFourHourly.setSouth(21);
        pm10TwentyFourHourly.setNorth(18);
        pm10TwentyFourHourly.setWest(17);
        pm10TwentyFourHourly.setNational(21);
        return pm10TwentyFourHourly;
    }

    protected PM10SubIndex getPM10SubIndex() {
        PM10SubIndex pm10SubIndex = new PM10SubIndex();
        pm10SubIndex.setEast(14);
        pm10SubIndex.setCentral(12);
        pm10SubIndex.setSouth(21);
        pm10SubIndex.setNorth(18);
        pm10SubIndex.setWest(17);
        pm10SubIndex.setNational(21);
        return pm10SubIndex;
    }

    protected PM25TwentyFourHourly getPM25TwentyFourHourly(){
        PM25TwentyFourHourly pm25TwentyFourHourly = new PM25TwentyFourHourly();
        pm25TwentyFourHourly.setEast(8);
        pm25TwentyFourHourly.setCentral(6);
        pm25TwentyFourHourly.setSouth(9);
        pm25TwentyFourHourly.setNorth(10);
        pm25TwentyFourHourly.setWest(7);
        pm25TwentyFourHourly.setNational(10);
        return pm25TwentyFourHourly;
    }

    protected COSubIndex getCOSubIndex(){
        COSubIndex coSubIndex = new COSubIndex();
        coSubIndex.setEast(4);
        coSubIndex.setCentral(5);
        coSubIndex.setSouth(2);
        coSubIndex.setNorth(4);
        coSubIndex.setWest(4);
        coSubIndex.setNational(5);
        return coSubIndex;
    }

    protected SO2SubIndex getSo2SubIndex(){
        SO2SubIndex so2SubIndex = new SO2SubIndex();
        so2SubIndex.setEast(8);
        so2SubIndex.setCentral(12);
        so2SubIndex.setSouth(12);
        so2SubIndex.setNorth(14);
        so2SubIndex.setWest(20);
        so2SubIndex.setNational(20);
        return so2SubIndex;
    }

    protected COEightHourMax getCOEightHourMax(){
        COEightHourMax coEightHourMax = new COEightHourMax();
        coEightHourMax.setEast(0.4);
        coEightHourMax.setCentral(0.48);
        coEightHourMax.setSouth(0.24);
        coEightHourMax.setNorth(0.43);
        coEightHourMax.setWest(0.42);
        coEightHourMax.setNational(0.48);
        return coEightHourMax;
    }

    protected NO2OneHourMax getNO2OneHourMax(){
        NO2OneHourMax no2OneHourMax = new NO2OneHourMax();
        no2OneHourMax.setEast(14);
        no2OneHourMax.setCentral(37);
        no2OneHourMax.setSouth(37);
        no2OneHourMax.setNorth(30);
        no2OneHourMax.setWest(35);
        no2OneHourMax.setNational(37);
        return no2OneHourMax;
    }

    protected SO2TwentyFourHourly getSO2TwentyFourHourly(){
        SO2TwentyFourHourly so2TwentyFourHourly = new SO2TwentyFourHourly();
        so2TwentyFourHourly.setEast(13);
        so2TwentyFourHourly.setCentral(19);
        so2TwentyFourHourly.setSouth(20);
        so2TwentyFourHourly.setNorth(23);
        so2TwentyFourHourly.setWest(32);
        so2TwentyFourHourly.setNational(32);
        return so2TwentyFourHourly;
    }

    protected PM25SubIndex getPM25SubIndex(){
        PM25SubIndex pm25SubIndex = new PM25SubIndex();
        pm25SubIndex.setEast(34);
        pm25SubIndex.setCentral(25);
        pm25SubIndex.setSouth(36);
        pm25SubIndex.setNorth(43);
        pm25SubIndex.setWest(30);
        pm25SubIndex.setNational(43);
        return pm25SubIndex;
    }

    protected PSITwentyFourHourly getPSITwentyFourHourly(){
        PSITwentyFourHourly psiTwentyFourHourly = new PSITwentyFourHourly();
        psiTwentyFourHourly.setEast(34);
        psiTwentyFourHourly.setCentral(25);
        psiTwentyFourHourly.setSouth(36);
        psiTwentyFourHourly.setNorth(43);
        psiTwentyFourHourly.setWest(30);
        psiTwentyFourHourly.setNational(43);
        return psiTwentyFourHourly;
    }

    protected O3EightHourMax getO3EightHourMax(){
        O3EightHourMax o3EightHourMax = new O3EightHourMax();
        o3EightHourMax.setEast(6);
        o3EightHourMax.setCentral(10);
        o3EightHourMax.setSouth(8);
        o3EightHourMax.setNorth(13);
        o3EightHourMax.setWest(9);
        o3EightHourMax.setNational(13);
        return o3EightHourMax;
    }

    public ArrayList<Region> getRegionList() {
        return regionList;
    }
}

File: app/src/androidTest/java/com/imdglobalservices/psi/PSISampleDataTest.kt
package com.imdglobalservices.psi

import com.imdglobalservices.psi.models.*
import java.util.*

/**
 * @author Dwi Setiyono <dwi.setiyono></dwi.setiyono>@ovo.id>
 * @since 2017.15.11
 */
open class PSISampleDataTest {

    val regionList = ArrayList<Region>()

    protected val o3SubIndexTest: O3SubIndex
        get() {
            val o3SubIndex = O3SubIndex()
            o3SubIndex.east = 3.0
            o3SubIndex.central = 4.0
            o3SubIndex.south = 3.0
            o3SubIndex.north = 5.0
            o3SubIndex.west = 4.0
            o3SubIndex.national = 5.0
            return o3SubIndex
        }

    protected val pM10TwentyFourHourly: PM10TwentyFourHourly
        get() {
            val pm10TwentyFourHourly = PM10TwentyFourHourly()
            pm10TwentyFourHourly.east = 14.0
            pm10TwentyFourHourly.central = 12.0
            pm10TwentyFourHourly.south = 21.0
            pm10TwentyFourHourly.north = 18.0
            pm10TwentyFourHourly.west = 17.0
            pm10TwentyFourHourly.national = 21.0
            return pm10TwentyFourHourly
        }

    protected val pM10SubIndex: PM10SubIndex
        get() {
            val pm10SubIndex = PM10SubIndex()
            pm10SubIndex.east = 14.0
            pm10SubIndex.central = 12.0
            pm10SubIndex.south = 21.0
            pm10SubIndex.north = 18.0
            pm10SubIndex.west = 17.0
            pm10SubIndex.national = 21.0
            return pm10SubIndex
        }

    protected val pM25TwentyFourHourly: PM25TwentyFourHourly
        get() {
            val pm25TwentyFourHourly = PM25TwentyFourHourly()
            pm25TwentyFourHourly.east = 8.0
            pm25TwentyFourHourly.central = 6.0
            pm25TwentyFourHourly.south = 9.0
            pm25TwentyFourHourly.north = 10.0
            pm25TwentyFourHourly.west = 7.0
            pm25TwentyFourHourly.national = 10.0
            return pm25TwentyFourHourly
        }

    protected val coSubIndex: COSubIndex
        get() {
            val coSubIndex = COSubIndex()
            coSubIndex.east = 4.0
            coSubIndex.central = 5.0
            coSubIndex.south = 2.0
            coSubIndex.north = 4.0
            coSubIndex.west = 4.0
            coSubIndex.national = 5.0
            return coSubIndex
        }

    protected val so2SubIndex: SO2SubIndex
        get() {
            val so2SubIndex = SO2SubIndex()
            so2SubIndex.east = 8.0
            so2SubIndex.central = 12.0
            so2SubIndex.south = 12.0
            so2SubIndex.north = 14.0
            so2SubIndex.west = 20.0
            so2SubIndex.national = 20.0
            return so2SubIndex
        }

    protected val coEightHourMax: COEightHourMax
        get() {
            val coEightHourMax = COEightHourMax()
            coEightHourMax.east = 0.4
            coEightHourMax.central = 0.48
            coEightHourMax.south = 0.24
            coEightHourMax.north = 0.43
            coEightHourMax.west = 0.42
            coEightHourMax.national = 0.48
            return coEightHourMax
        }

    protected val nO2OneHourMax: NO2OneHourMax
        get() {
            val no2OneHourMax = NO2OneHourMax()
            no2OneHourMax.east = 14.0
            no2OneHourMax.central = 37.0
            no2OneHourMax.south = 37.0
            no2OneHourMax.north = 30.0
            no2OneHourMax.west = 35.0
            no2OneHourMax.national = 37.0
            return no2OneHourMax
        }

    protected val sO2TwentyFourHourly: SO2TwentyFourHourly
        get() {
            val so2TwentyFourHourly = SO2TwentyFourHourly()
            so2TwentyFourHourly.east = 13.0
            so2TwentyFourHourly.central = 19.0
            so2TwentyFourHourly.south = 20.0
            so2TwentyFourHourly.north = 23.0
            so2TwentyFourHourly.west = 32.0
            so2TwentyFourHourly.national = 32.0
            return so2TwentyFourHourly
        }

    protected val pM25SubIndex: PM25SubIndex
        get() {
            val pm25SubIndex = PM25SubIndex()
            pm25SubIndex.east = 34.0
            pm25SubIndex.central = 25.0
            pm25SubIndex.south = 36.0
            pm25SubIndex.north = 43.0
            pm25SubIndex.west = 30.0
            pm25SubIndex.national = 43.0
            return pm25SubIndex
        }

    protected val psiTwentyFourHourly: PSITwentyFourHourly
        get() {
            val psiTwentyFourHourly = PSITwentyFourHourly()
            psiTwentyFourHourly.east = 34.0
            psiTwentyFourHourly.central = 25.0
            psiTwentyFourHourly.south = 36.0
            psiTwentyFourHourly.north = 43.0
            psiTwentyFourHourly.west = 30.0
            psiTwentyFourHourly.national = 43.0
            return psiTwentyFourHourly
        }

    protected val o3EightHourMax: O3EightHourMax
        get() {
            val o3EightHourMax = O3EightHourMax()
            o3EightHourMax.east = 6.0
            o3EightHourMax.central = 10.0
            o3EightHourMax.south = 8.0
            o3EightHourMax.north = 13.0
            o3EightHourMax.west = 9.0
            o3EightHourMax.national = 13.0
            return o3EightHourMax
        }

    protected fun setTestData(lat: Double, lng: Double, name: String) {
        val region = Region()
        val location = Location()
        location.latitude = lat
        location.longitude = lng
        region.name = name
        region.labelLocation = location
        regionList.add(region)
    }
}
